api/turing/validation/validator.go

```diff
@@ -196,6 +196,81 @@ func checkDanglingRoutes(
 	}
 }
 
+func validateConditionOrthogonality(
+	sl validator.StructLevel,
+	fieldName string,
+	allRules models.TrafficRules,
+) {
+	trafficConditionsFieldMap := map[string][]string{}
+	trafficConditionsValueMap := map[string]map[string]*set.Set{}
+	for _, rule := range allRules {
+		trafficConditionsValueMap[rule.Name] = map[string]*set.Set{}
+		trafficConditionsFieldMap[rule.Name] = []string{}
+		for _, condition := range rule.Conditions {
+			conditionValues := []interface{}{}
+			for _, val := range condition.Values {
+				conditionValues = append(conditionValues, val)
+			}
+			trafficConditionsValueMap[rule.Name][condition.Field] = set.New(conditionValues...)
+			trafficConditionsFieldMap[rule.Name] = append(trafficConditionsFieldMap[rule.Name], condition.Field)
+		}
+	}
+
+	allErrors := []string{}
+	for idx, rule1 := range allRules {
+		rule1Fields, ok := trafficConditionsFieldMap[rule1.Name]
+		// Check that there are rule conditions
+		if ok {
+			// Check that the current rule and the other are orthogonal
+			rulesOverlap := true
+
+			for _, rule2 := range allRules[idx+1:] {
+				// Rules with same name should fail unique name validation
```

Comment:
Is this comment correct? I think you mean to say "should fail the overlapping rules validation" ?

Reply:
Oh, yes it's correct. What I meant to say here was that this case can be skipped for orthogonality checks, because it would have failed the unique name check anyway, and it doesn't make sense to check the same rule against itself.